### PR TITLE
Vendoring libnetwork @bdc9dce 

### LINKED ISSUE #25393: Failed to create network - possible concurrency issue?
**Output of `docker version`:**

```
Client:
Version:      1.11.1
API version:  1.23
Go version:   go1.5.4
Git commit:   5604cbe
Built:        Tue Apr 26 23:30:23 2016
OS/Arch:      linux/amd64

Server:
Version:      1.12.0
API version:  1.24
Go version:   go1.6.3
Git commit:   8eab29e
Built:
OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 41
Running: 38
Paused: 0
Stopped: 3
Images: 437
Server Version: 1.12.0
Storage Driver: devicemapper
Pool Name: docker-253:0-204930528-pool
Pool Blocksize: 65.54 kB
Base Device Size: 10.74 GB
Backing Filesystem: xfs
Data file: /dev/dcspa14l_appvg/docker_data
Metadata file: /dev/dcspa14l_appvg/docker_metadata
Data Space Used: 116 GB
Data Space Total: 214.7 GB
Data Space Available: 98.76 GB
Metadata Space Used: 129 MB
Metadata Space Total: 5.369 GB
Metadata Space Available: 5.24 GB
Thin Pool Minimum Free Space: 21.47 GB
Udev Sync Supported: true
Deferred Removal Enabled: false
Deferred Deletion Enabled: false
Deferred Deleted Device Count: 0
Library Version: 1.02.107-RHEL7 (2015-10-14)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
Volume: local
Network: bridge null host overlay
Kernel Version: 3.10.0-327.el7.x86_64
Operating System: Red Hat Enterprise Linux Server 7.2 (Maipo)
OSType: linux
Architecture: x86_64
CPUs: 24
Total Memory: 62.73 GiB
Name: dcspa14l.unix.anz
ID: FWD6:2DYT:2SPX:PJ4T:5NPF:NYX3:LJGE:5AMT:XOY3:ZBWU:N7EP:ZXVG
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Machine is physical hardware, AMD Opteron(tm) Processor 6174, 24 cores/cpu, 64GB RAM. Operating system is Redhat 7.2.  Dockerproject is only 3rd-party RPMs. Docker running natively on host.  /var/run/docker.sock exposed to (bamboo agent) containers via volume mount, actual "docker" commands running in the containers.  This works 100% fine on previous (VM) hosts, but now having occasional errors as described below.  

**Steps to reproduce the issue:**
1. System runs 24/7.  Multiple bamboo agents (each in docker containers) pick up jobs, and execute CI-type activities: building code etc, before spinning up an environment using docker-compose.  /var/run/docker.sock is volume-mapped from host to each guest.
2. A couple of times a day (maybe 1%) we get error below, and network fails to create
3. Each of the 5 bamboo agents should create their own separate network a1_default to a5_default

**Describe the results you received:**

```
03-Aug-2016 18:23:14 + set -e
03-Aug-2016 18:23:14 + docker network ls
03-Aug-2016 18:23:14 NETWORK ID          NAME                DRIVER
03-Aug-2016 18:23:14 99205c0d2a4d        agents_default      bridge             
03-Aug-2016 18:23:14 d538f34a8c1b        bridge              bridge             
03-Aug-2016 18:23:14 f1d1f3e1c863        host                host               
03-Aug-2016 18:23:14 a5a8158b87e3        none                null               
03-Aug-2016 18:23:14 + docker-compose down
03-Aug-2016 18:23:16 Removing network a4_default
03-Aug-2016 18:23:16 Network a4_default not found.
03-Aug-2016 18:23:16 + echo 'docker-compose down returned 0'
03-Aug-2016 18:23:16 + docker network ls
03-Aug-2016 18:23:16 docker-compose down returned 0
03-Aug-2016 18:23:16 NETWORK ID          NAME                DRIVER
03-Aug-2016 18:23:16 99205c0d2a4d        agents_default      bridge             
03-Aug-2016 18:23:16 d538f34a8c1b        bridge              bridge             
03-Aug-2016 18:23:16 f1d1f3e1c863        host                host               
03-Aug-2016 18:23:16 a5a8158b87e3        none                null               
03-Aug-2016 18:23:16 + sleep 5
03-Aug-2016 18:23:21 + docker-compose pull nginx simulator redis activemq oracle
03-Aug-2016 18:23:23 Pulling redis (csp-docker-registry.service.dev:5000/redis:latest)...
03-Aug-2016 18:23:23 latest: Pulling from redis
03-Aug-2016 18:23:23 Digest: sha256:eac6e2dec2f645d110b4013f1e6a59fa953a54d46fc5400172b7433fa0738593
03-Aug-2016 18:23:23 Status: Image is up to date for csp-docker-registry.service.dev:5000/redis:latest
03-Aug-2016 18:23:23 Pulling oracle (csp-docker-registry.service.dev:5000/oracle-12c:latest)...
03-Aug-2016 18:23:23 latest: Pulling from oracle-12c
03-Aug-2016 18:23:23 Digest: sha256:883722fd7a6c30c5f6c992104f0c29f4c9409737fef71eaac7f405000e75328e
03-Aug-2016 18:23:23 Status: Image is up to date for csp-docker-registry.service.dev:5000/oracle-12c:latest
03-Aug-2016 18:23:23 Pulling activemq (csp-docker-registry.service.dev:5000/rmohr/activemq:5.11.1)...
03-Aug-2016 18:23:23 5.11.1: Pulling from rmohr/activemq
03-Aug-2016 18:23:23 Digest: sha256:aebe8eeaf339ef4d5406b7881e87f939949a2311f69a7e4a11d4a21900d813db
03-Aug-2016 18:23:23 Status: Image is up to date for csp-docker-registry.service.dev:5000/rmohr/activemq:5.11.1
03-Aug-2016 18:23:23 Pulling simulator (csp-docker-registry.service.dev:5000/anz/simulator-develop:latest)...
03-Aug-2016 18:23:23 latest: Pulling from anz/simulator-develop
03-Aug-2016 18:23:45 Digest: sha256:55b64adf97f31c417888ba33b68d9e80d8f6bd4e6a9497ccc726add8fd6951ac
03-Aug-2016 18:23:45 Status: Downloaded newer image for csp-docker-registry.service.dev:5000/anz/simulator-develop:latest
03-Aug-2016 18:23:45 Pulling nginx (csp-docker-registry.service.dev:5000/nginx:latest)...
03-Aug-2016 18:23:45 latest: Pulling from nginx
03-Aug-2016 18:23:45 Digest: sha256:9671815a4cfa4d9b64ff808abe82b5cce895d9391511f3471270cb1a57753cf1
03-Aug-2016 18:23:45 Status: Image is up to date for csp-docker-registry.service.dev:5000/nginx:latest
03-Aug-2016 18:23:45 + docker-compose up -d --force-recreate nginx simulator redis activemq oracle
03-Aug-2016 18:23:47 Creating network "a4_default" with the default driver
03-Aug-2016 18:23:48 unable to remove jump to DOCKER-ISOLATION rule in FORWARD chain:  (iptables failed: iptables --wait -D FORWARD -j DOCKER-ISOLATION: iptables: No chain/target/match by that name.
03-Aug-2016 18:23:48 (exit status 1))
```

**Describe the results you expected:**
Network a4_default created per normal (and all other times works OK)

**Additional information you deem important (e.g. issue happens only occasionally):**

Main error is:

```
unable to remove jump to DOCKER-ISOLATION rule in FORWARD chain:  (iptables failed: iptables --wait -D FORWARD -j DOCKER-ISOLATION: iptables: No chain/target/match by that name.
```

Prior to 1.12.0 was getting errors like:

```
Jul 25 15:23:39 dcspa10l docker: time="2016-07-25T15:23:39.372045790+10:00" level=error msg="Handler for POST /v1.22/networks/create returned error: failed to parse pool request for address space \"LocalDefault\" pool \"\" subpool \"\": could not find an available predefined network"
```

Which looks a LOT like https://github.com/docker/libnetwork/issues/1101  !!   Haven't seen that since updated to 1.12.0, removed `/var/lib/docker/network/*` and rebooted.

**********\* Perhaps issue is concurrency?  There are FIVE bamboo agents, each of which could possibly be accessing docker at the same time.    ***********

Also noticed a race condition (??) between "docker-compose down" and subsequent "docker-compose up" - added delay.

Also had to disable network-manager completely, to stop lots of noise (??) with it noticing the interfaces.

Also lots of noise in dmesg like:

```
[271912.501730] XFS (dm-18): Mounting V4 Filesystem
[271912.598299] XFS (dm-18): Ending clean mount
[271912.599781] device veth866953f entered promiscuous mode
[271912.599897] IPv6: ADDRCONF(NETDEV_UP): veth866953f: link is not ready
[271912.599901] br-b5841fcebac4: port 4(veth866953f) entered forwarding state
[271912.599904] br-b5841fcebac4: port 4(veth866953f) entered forwarding state
[271912.600341] br-b5841fcebac4: port 4(veth866953f) entered disabled state
[271912.816148] IPv6: ADDRCONF(NETDEV_CHANGE): veth866953f: link becomes ready
[271912.816214] br-b5841fcebac4: port 4(veth866953f) entered forwarding state
[271912.816218] br-b5841fcebac4: port 4(veth866953f) entered forwarding state
[271913.095693] XFS (dm-19): Mounting V4 Filesystem
[271913.206525] XFS (dm-19): Ending clean mount
```

### PR DESCRIPTION
**Pull request by aboch**
Fixes #25393

### CODE REVIEW

**Comment by vieux** [comment]
LGTM ping @tiborvass

File: vendor/github.com/docker/libnetwork/network.go [hunk 1]
```diff
@@ -1118,6 +1118,8 @@ func (n *network) addSvcRecords(name string, epIP net.IP, epIPv6 net.IP, ipMapUp
 		return
 	}
 
+	logrus.Debugf("(%s).addSvcRecords(%s, %s, %s, %t)", n.ID()[0:7], name, epIP, epIPv6, ipMapUpdate)
```
Comment (tiborvass) [inline vendor/github.com/docker/libnetwork/network.go]: was this meant to stay?

Comment (aboch) [inline vendor/github.com/docker/libnetwork/network.go]: Yes, we need it to debug some weird bug around missing service records.

**Comment by tiborvass** [comment]
LGTM

**Comment by thaJeztah** [comment]
associated libnetwork PR; https://github.com/docker/libnetwork/pull/1658

**Comment by thaJeztah** [comment]
Libnetwork cherry-pick for 17.03.2 https://github.com/docker/libnetwork/pull/1750

**Pull request merged**